mitosheet/src/plugin.tsx

```diff
@@ -160,12 +165,23 @@ function activateMitosheetExtension(
             const activeCellIndex = notebook.activeCellIndex;
 
             const codeCell = getCellAtIndex(cells, mitosheetCallIndex + 1);
+            const codeCellText = getCellText(codeCell);
 
-            if (isEmptyCell(codeCell) || containsGeneratedCodeOfAnalysis(getCellText(codeCell), analysisName)) {
+
+            // If the user has edited the code and they haven't chosen whether or not to overwrite the contents of the cell,
+            // trigger the dialog to ask them. 
+            if (overwriteIfUserEditedCode === undefined && !isEmptyCell(codeCell) && hasCodeCellBeenEditedByUser(oldCode, codeCellText)) {
+                triggerUserEditedCodeDialog();
+                return;
+            // Only write to the cell if either of the following are true:
+            // 1. The user has authorized overwriting the cell
+            // 2. The cell hasn't been edited by the user
+            // AND the cell exists. If the cell doesn't exist we can't write to it!
+            } else if (codeCell !== undefined && (overwriteIfUserEditedCode || !hasCodeCellBeenEditedByUser(oldCode, codeCellText))) {
                 writeToCell(codeCell, code)
```

Comment:
What happened to the previous check here around containsGeneratedCodeOfAnalysis? Are we just missing this now / do we not need it?

I think we need to be really careful around this logic, as it's so complex. You did a good job simplifying since the last PR, but it's still not _obviously_ correct to me -- it take some reading.

Reply:
Oh yeah I left a comment in that PR about this - I don't think we need that anymore because we're already checking that it matches the code we generate. After looking into what that function was doing, it seemed like it might interfere with our checks (i.e. if the user made some change that made it look like it wasn't our code, it would just automatically create another cell). Unless you have another case that would explain a need to put that back in, I think the current check is sufficient. Lmk!